source/std.tex

```diff
@@ -49,6 +49,15 @@
            ]{hyperref}
 \usepackage{memhfixc}     % fix interactions between hyperref and memoir
 
+
+%% Grammar extraction.
+\IfFileExists{./std-gram.ext}{
+\usepackage[extract-env={bnftab,simplebnf,bnf,bnfkeywordtab}]{extract}
+\def\PrintGrammar{\input{std-gram.ext}}}{
+\usepackage[active,header=false,handles=false,copydocumentclass=false,generate=std-gram.ext,extract-env={bnftab,simplebnf,bnf,bnfkeywordtab}]{extract}
+\def\PrintGrammar{Run again to include grammar summary}
```

Comment:
Do we really need this two-pass approach? Is there no way to flush the grammar at the end of the normal clauses so we can do this in one go? It seems easier to get this wrong than the current approach, at least in some ways. If we can't make this automatic, then at least we should use something like \ref{grammar.not.extracted.yet} here to trigger latexmk to do another iteration, and we should always regenerate the file even if we already have a copy.